file_browser/src/components/AddDIalog.tsx

```diff
@@ -0,0 +1,106 @@
+import { Dispatch, SetStateAction, useState } from "react";
+import { SystemItem } from "../classes/SystemItem";
+import { FileClass } from "../classes/File";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+
+const options = ["File", "Folder"];
+
+function AddDialog({
+  setDialog,
+}: {
+  setDialog: Dispatch<SetStateAction<boolean>>;
+}) {
+  const [selected, setSelected] = useState<any>(0);
+  const [name, setName] = useState("");
+  const [content, setContent] = useState("");
+  const [error, setError] = useState("");
+
+  const handleAdd = (item: SystemItem) => {
+    (fileSystemStore.selectedItem as Folder).add(item);
+  };
+
+  function onChange(i: any) {
+    setSelected((prev: any) => (i === prev ? null : i));
+    setError("")
+  }
+
+  function createNewItem() {
+    let item;
```

Comment:
In TypeScript when you define a variable, you should define a type constraint for it. In this case it'd be `SystemItem`